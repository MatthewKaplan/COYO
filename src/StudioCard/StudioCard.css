
*,
*:after,
*:before {
  box-sizing: border-box;
}

.address-block {
  display: block;
}

a {
  text-decoration: none;
  color: inherit;
}




.studioImgs {
  position: relative;
  filter: brightness(60%);
  max-height: 120px;
  object-fit: cover; 
  width: 100%;
  z-index: 0;
}

.studioName {
  font-family: 'Aladin', cursive;
  position: absolute;
  font-size: 45px;
  color: #F4F9E9;
  top: 0;
  left:0;
  z-index: 2;
  padding: .5rem 0 0 .5rem;
}

.studioFee {
  font-family: 'Raleway', sans-serif;
   position: absolute;
  top: 5px;
  z-index: 3;
  right: 1rem;
  font-size: 23px;
  text-align: center;
  
  color: #F4F9E9;
  text-transform: uppercase;
}

.drop-in-text {
  letter-spacing: 1px;
/*   font-family: 'Aladin', cursive; */
  font-size: 13px;
}

/* .studioCard {
  text-align: center;
  padding: 1rem;
   font-family: 'Raleway', sans-serif;
} */



.studioNumber {
  font-size: 20px;
/*   padding: 3px; */
}

.studioLink {
  overflow-wrap: break-word;
  transition:all 0.4s ease;
  font-size: 14px;
  background: lightgrey;
  text-transform: uppercase;
  padding: 5px;
  margin-top: 5px;
  letter-spacing: 3px;
  border-radius: 5px;
}

.studio-link:hover {
  color: purple;
  font-weight: bold;
  background: white;
}

.mission-container {
  text-align: left;
}

.mission-heading {
  font-weight: bold;
  margin: 1rem 0 0 0;
  text-transform: uppercase;
}

.studioRating {
  margin-left: 10px;
}

.inline {
  display: flex;
  justify-content: center;
}

.starsOuter {
  position: relative;
  display: inline-block;
}

.starsInner {
  position: absolute;
  top: 0;
  left: 0;
  white-space: nowrap;
  overflow: hidden;
  width: 0;
}

.starsInner::before {
  content: "\f005 \f005 \f005 \f005 \f005";
  font-family: 'Font Awesome 5 Free';
  font-weight: 900;
  color: #f8ce0b;
}

.starsOuter::before {
  content: "\f005 \f005 \f005 \f005 \f005";
  font-family: 'Font Awesome 5 Free';
  font-weight: 900;
  color: #ccc;
}

.studioMission {
  margin: .5rem 0 1rem 0;
  height: 100px;
  overflow: auto;
}

