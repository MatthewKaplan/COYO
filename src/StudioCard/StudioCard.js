import React, { Component } from 'react'
import '../Styles/_studioCard.scss';

export default class StudioCard extends Component {
  constructor(props) {
    super(props);
    this.state = {

    }
  }

  render() {
    const starsInnerStyle = {
    width: this.props.roundedRating
    }

  return (
      <article className="studioCard">
        <header className="cardCover">
          <img src={this.props.image} alt="Group Yoga in studio" className="studioImg"/>
          <h4 className="studioName">{this.props.name}</h4>
          <div className="studioFee">
            {this.props.fee} 
            <div className="dropInText">drop in</div>
          </div>
        </header>
        <main className="studioInfo">
          <address className="contactInfo">
            <p className="address">{this.props.address}</p>
            <a className="phoneNum" href="tel:3034162958">{this.props.phone}</a>      
          </address>
          <div className="missionContainer">
            <h4 className="missionHeading">Mission:</h4>
            <p className="studioMission">{this.props.mission}</p>
          </div>
          <div className="inline">
            <div className="starsOuter">
              <div className="starsInner" style={starsInnerStyle}></div>
          </div>
            <p className="studioRating">{this.props.rating}</p>
          </div>
          <a href={this.props.link} className="studioLink" rel="noopener noreferrer" target="_blank">Learn More</a>
        </main>
      </article>
    )
  }
}

