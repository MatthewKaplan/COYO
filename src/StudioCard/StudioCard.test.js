import React from 'react';
import StudioCard from './studioCard';
import { shallow } from 'enzyme';

const mockStudio = {
  address: "1735 E 17th Ave, Denver, CO 80218",
  dropInFee: "$17.00",
  googleRating: 4.8,
  img: "https://i.imgur.com/w7KfQWv.jpg",
  link: "https://wholeyoga.com",
  mainTypeOffered: 5,
  mission: "Where body and mind become whole.",
  name: "Whole Yoga",
  openEveryday: true,
  phoneNum: "(303)333-9642",
  teacherTraining: true
}

describe('StudioCard', () => {
  let wrapper
  beforeEach(() => {
  wrapper = shallow(
   <StudioCard  image={mockStudio.img}
                name={mockStudio.name}
                address={mockStudio.address}
                phone={mockStudio.phoneNum}
                mission={mockStudio.mission}
                rating={mockStudio.googleRating}
                roundedRating={`${Math.round(mockStudio.googleRating / 5 * 100 / 10) * 10}%`}
                link={mockStudio.link}
                fee={mockStudio.dropInFee} />
      )            
  });

  it('should render correctly with all the information passed', () => {
      expect(wrapper).toMatchSnapshot();
    });

    
});