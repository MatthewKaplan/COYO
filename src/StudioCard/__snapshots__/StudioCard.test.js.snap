// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudioCard should match snapshot 1`] = `
<article
  className="studioCard"
>
  <header
    className="cardCover"
  >
    <img
      alt="Group Yoga in studio"
      className="studioImg"
    />
    <h4
      className="studioName"
    />
    <div
      className="studioFee"
    >
      <div
        className="dropInText"
      >
        drop in
      </div>
    </div>
  </header>
  <main
    className="studioInfo"
  >
    <address
      className="contactInfo"
    >
      <p
        className="address"
      />
      <a
        className="phoneNum"
        href="tel:3034162958"
      />
    </address>
    <div
      className="missionContainer"
    >
      <h4
        className="missionHeading"
      >
        Mission:
      </h4>
      <p
        className="studioMission"
      />
    </div>
    <div
      className="inline"
    >
      <div
        className="starsOuter"
      >
        <div
          className="starsInner"
          style={
            Object {
              "width": undefined,
            }
          }
        />
      </div>
      <p
        className="studioRating"
      />
    </div>
    <a
      className="studioLink"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn More
    </a>
  </main>
</article>
`;

exports[`StudioCard should render correctly with all the information passed 1`] = `
<article
  className="studioCard"
>
  <header
    className="cardCover"
  >
    <img
      alt="Group Yoga in studio"
      className="studioImg"
      src="https://i.imgur.com/w7KfQWv.jpg"
    />
    <h4
      className="studioName"
    >
      Whole Yoga
    </h4>
    <div
      className="studioFee"
    >
      $17.00
      <div
        className="dropInText"
      >
        drop in
      </div>
    </div>
  </header>
  <main
    className="studioInfo"
  >
    <address
      className="contactInfo"
    >
      <p
        className="address"
      >
        1735 E 17th Ave, Denver, CO 80218
      </p>
      <a
        className="phoneNum"
        href="tel:3034162958"
      >
        (303)333-9642
      </a>
    </address>
    <div
      className="missionContainer"
    >
      <h4
        className="missionHeading"
      >
        Mission:
      </h4>
      <p
        className="studioMission"
      >
        Where body and mind become whole.
      </p>
    </div>
    <div
      className="inline"
    >
      <div
        className="starsOuter"
      >
        <div
          className="starsInner"
          style={
            Object {
              "width": "100%",
            }
          }
        />
      </div>
      <p
        className="studioRating"
      >
        4.8
      </p>
    </div>
    <a
      className="studioLink"
      href="https://wholeyoga.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn More
    </a>
  </main>
</article>
`;
